src/content-helper/dashboard-widget/provider.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ContentHelperError,
+	ContentHelperErrorCode,
+} from '../../blocks/content-helper/content-helper-error';
+import { TopPostData } from './top-posts/model';
+import {
+	convertDateToString,
+	removeDaysFromDate,
+} from '../../blocks/shared/utils/date';
+
+/**
+ * The form of the response returned by the /stats/posts WordPress REST API
+ * endpoint.
+ */
+interface TopPostsApiResponse {
+	error?: Error;
+	data?: TopPostData[];
+}
+
+/**
+ * The form of the result returned by the getTopPosts() function.
+ */
+export interface GetTopPostsResult {
+	posts: TopPostData[];
+}
+
+export const TOP_POSTS_DEFAULT_LIMIT = 3;
+export const TOP_POSTS_DEFAULT_TIME_RANGE = 7; // In days.
+
+class TopPostsProvider {
+	private dataPeriodStart: string;
+	private dataPeriodEnd: string;
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		this.dataPeriodEnd = convertDateToString( new Date() ) + 'T23:59';
+		this.dataPeriodStart = removeDaysFromDate(
+			this.dataPeriodEnd,
+			TOP_POSTS_DEFAULT_TIME_RANGE - 1
+		) + 'T00:00';
```

Comment:
Using same logic inside `PerformanceDetailsProvider` .... should we make a common function for it?

Reply:
Yes, but not in this PR. See [this](https://github.com/Parsely/wp-parsely/pull/1305#discussion_r1086950059).